ComplexTrStructure/ComplexTrStructure/ComplexTr.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ComplexTrStructure
+{
+    public class ComplexTr
+    {
+        double abs;
+
+        public double Abs
+        {
+            get => abs;
+
+            set
+            {
+                if (value < 0)
+                    throw new ArgumentException("Значение должно быть неотрицательным");
+                abs = value;
+            }
+        }
+        double arg;
+        public double Arg
+        {
+            get => arg;
+            set => arg = value;
+        }
+        public double Re
+        { 
+            get => Math.Round(Abs* Math.Cos(arg), 13);
+        }
+
+        public double Im
+        {
+            get => Math.Round(Abs * Math.Sin(arg), 13);
+        }
+
+        public ComplexTr(double abs, double arg)
+        {
+            this.Abs = abs;  
+            this.Arg = arg;  
+        }
+
+
+        public override string ToString()
+        {
+            if (Abs == 0)
+                return $"Cos({Arg}) + i Sin({Arg})";
+            else 
+                return $"{Abs} (Cos({Arg}) + i Sin({Arg}))";
+        }
+
+        public override bool Equals(object obj)
+        {
+           if (obj is ComplexTr)
+                return (Abs == ((ComplexTr)obj).Abs) && ((Arg - ((ComplexTr)obj).Arg) % (Math.PI * 2) == 0);
```

Comment:
При вычислениях с плавающей точкой правильнее проводить сравнения не абсолютно, а с заданной точностью (в задании 10^(-13))
```suggestion
                return (Abs == ((ComplexTr)obj).Abs) && (Math.Abs((Arg - ((ComplexTr)obj).Arg) % (Math.PI * 2)) < 1e-13);
```